### PR TITLE
Introduce Script_Data

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue #1004 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Core/Assets/Assets.php [hunk 1]
```diff
@@ -343,15 +343,34 @@ private function get_assets() {
...
+					'global'        => 'googlesitekit',
+					'data_callback' => function () {
+						return $this->get_inline_data();
+					},
+				)
+			),
+			new Script_Data(
+				'googlesitekit-base-data',
+				array(
+					'global'        => '_googlesitekitBase',
```
Comment (felixarntz) [inline includes/Core/Assets/Assets.php]: We should probably define a pattern as part of this PR on how to name these globals. Right now the two defined here (`googlesitekit` can be ignored, as it's old) are different as in only one has a `Data` suffix.

Maybe we should set the following rules:
* Every `Script_Data` should have a handle that ends with `-data`.
* Every `Script_Data` should use a global name that matches the handle transformed into camel-case, and prefixed with an underscore (e.g. `googlesitekit-base-data` becomes `_googlesitekitBaseData`).

What do you think?

Comment (felixarntz) [reply includes/Core/Assets/Assets.php]: If you agree, let's change the corresponding JS references as well of course.

Comment (aaemnnosttv) [reply includes/Core/Assets/Assets.php]: That sounds good, I'll update those 👍

Comment (aaemnnosttv) [reply includes/Core/Assets/Assets.php]: Not too many replacements needed.  ✅ https://github.com/google/site-kit-wp/pull/1189/commits/081c145305c843eeb0afd56e5eddb14b2ad60aa7

**Pull request merged**